src/solver/any/forward_any.cpp

```diff
@@ -0,0 +1,338 @@
+    auto size        = ((input_numel + LOCAL_SIZE - 1) / LOCAL_SIZE);
+
+    auto dtype = inputDesc.GetType();
+    size *= get_data_size(dtype);
+    size_t data_size      = get_data_size(dtype);
+    size_t workspace_size = AlignUp(size, LOCAL_SIZE) / LOCAL_SIZE;
+    size_t ws_scratch_mem = 2 * workspace_size * data_size;
+    size_t ws_local_mem   = LOCAL_SIZE * data_size;
+
+    return MultiBufferWorkspaceTraits{ws_scratch_mem, ws_local_mem};
+}
+
+bool AnyForward::IsApplicable(const ExecutionContext& context,
+                              const miopen::any::ProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    return true;
+}
+
+ConvSolution AnyForward::GetSolution(const ExecutionContext& context,
+                                     const miopen::any::ProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype        = problem.GetInputDesc().GetType();
+    auto input_dtype  = miopen::GetDataType(problem.GetInputDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetOutputDesc().GetType());
+
+    std::cout << "input_dtype: " << input_dtype << std::endl;
+
+    auto input_dims  = problem.GetInputDesc().GetLengths();
+    auto output_dims = problem.GetOutputDesc().GetLengths();
+    auto dim         = problem.GetDim();
+
+    auto input_numel  = problem.GetInputDesc().GetElementSize();
+    auto output_numel = problem.GetOutputDesc().GetElementSize();
+
+    // auto in_dtype = input_dtype;
+    // if(in_dtype == "int8_t")
+    // {
+    //     in_dtype = "signed char";
+    // }
+    // else if(in_dtype == "bfloat16")
+    // {
+    //     in_dtype = "ushort";
+    // }
+
+    std::cout << "input_dtype: " << input_dtype << std::endl;
+
+    if(dim != -1)
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(output_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel        = KernelInfo{};
+        kernel.kernel_file = "MIOpenAny.cpp";
+        kernel.kernel_name = "AnyForward";
+
+        KernelBuildParameters build_params;
+
+        // Awkward way to handle different input types
+        // TODO: Find other elegant way to handle this
+        if(input_dtype == "int8_t")
+        {
+            build_params = KernelBuildParameters{
+                {"MIOPEN_USE_INT8", static_cast<int>(dtype == miopenInt8)},
+            };
+        }
+        else
+        {
+            build_params = KernelBuildParameters{
+                {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+                {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+                {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+                {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype}};
+        }
```

Comment:
Can we do like this? Also as discussed internal, I am investigating why KernelBuildParameters cannot pass string with a space
```suggestion
            build_params = KernelBuildParameters{
                {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
                {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
                {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
                {"MIOPEN_USE_INT8", static_cast<int>(dtype == miopenInt8)},
                {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype == "int8_t" ? "signed char" : input_dtype}};
        
```

Reply:
For this issue, I realized that we can simply solve it as following: 

Instead of manually setting `{"INPUT_TYPE", "signed char"}`, we can:

1. Use the `MIOPEN_USE_INT8` macro
2. Include `<miopen_cstdint.hpp>` in the kernel implementation file

This approach automatically handles `INPUT_DTYPE=int8_t` and avoids a bug where `BuildParams` incorrectly parses string data types containing spaces (it only captures the last word instead of the complete string).